make/autoconf/jdk-version.m4

```diff
@@ -69,34 +69,38 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   AC_SUBST(JDK_RC_PLATFORM_NAME)
   AC_SUBST(HOTSPOT_VM_DISTRO)
 
+  # Note: UTIL_ARG_WITH treats empty strings as valid values when OPTIONAL is false!
+  # That said, it allows for using the RESULT of an option as its own DEFAULT, which
+  # is pretty convenient...
+
+  # Outer [ ] to quote m4.
+  [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
+
   # Setup username (for use in adhoc version strings etc)
-  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
-      [build username to use in version strings])])
-  if test "x$with_build_user" = xyes || test "x$with_build_user" = xno; then
-    AC_MSG_ERROR([--with-build-user must have a value])
-  elif test "x$with_build_user" != x; then
-    USERNAME="$with_build_user"
-  else
-    # Outer [ ] to quote m4.
-    [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
-  fi
+  UTIL_ARG_WITH(NAME: build-user, TYPE: string,
+    RESULT: USERNAME,
+    DEFAULT: $USERNAME,
+    DESC: [build username to use in version strings],
+    IF_GIVEN: [
+      if test "x$USERNAME" = x; then
+        AC_MSG_ERROR([--with-build-user cannot be empty])
+      fi
+    ])
   AC_SUBST(USERNAME)
 
   # Set the JDK RC name
-  AC_ARG_WITH(jdk-rc-name, [AS_HELP_STRING([--with-jdk-rc-name],
-      [Set JDK RC name. This is used for FileDescription and ProductName properties
-       of MS Windows binaries. @<:@not specified@:>@])])
-  if test "x$with_jdk_rc_name" = xyes || test "x$with_jdk_rc_name" = xno; then
-    AC_MSG_ERROR([--with-jdk-rc-name must have a value])
-  elif [ ! [[ $with_jdk_rc_name =~ ^[[:print:]]*$ ]] ]; then
-    AC_MSG_ERROR([--with-jdk-rc-name contains non-printing characters: $with_jdk_rc_name])
-  elif test "x$with_jdk_rc_name" != x; then
-    # Set JDK_RC_NAME to a custom value if '--with-jdk-rc-name' was used and is not empty.
-    JDK_RC_NAME="$with_jdk_rc_name"
-  else
-    # Otherwise calculate from "branding.conf" included above.
-    JDK_RC_NAME="$PRODUCT_NAME $JDK_RC_PLATFORM_NAME"
-  fi
+  # Otherwise calculate from "branding.conf" included above.
+  UTIL_ARG_WITH(NAME: jdk-rc-name, TYPE: string,
+    DEFAULT: [$PRODUCT_NAME $JDK_RC_PLATFORM_NAME],
+    DESC: [Set JDK RC name. This is used for FileDescription and ProductName
```

Comment:
Embarrassingly I left it that way since I didn't know what `@<:@...@:>@` actually did. So the following replacement should work?
DEFAULT_DESC: [Set JDK RC name. This is used for FileDescription and ProductName properties of MS Windows binaries.]

Reply:
I've changed them accordingly to what should be right, do let me know if I've made any mistakes in my usage of `DEFAULT_DESC`